mule-extensions-api/src/main/java/org/mule/runtime/extension/api/notification/NotificationHandler.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.notification;
+
+import static org.mule.runtime.extension.api.notification.ExtensionNotification.registerAction;
+
+/**
+ * This handler allows firing {@link org.mule.runtime.api.notification.Notification Notifications} with both custom data and
+ * default one.
+ *
+ * @since 1.1
+ */
+public interface NotificationHandler {
+
+  /**
+   * Registers a given notification action with it's name. All extension actions should be in the
+   * {@link org.mule.runtime.api.notification.AbstractServerNotification#EXTENSION_EVENT_ACTION_START_RANGE} range.
+   *
+   * @param name the name for the action.
+   * @param action the code for the action.
+   */
+  static void register(String name, int action) {
```

Comment:
this shouldn't exist. Can be auto generate the action code? We could replace this ugly int with an enum and then do BASE_ACTION + enum.hashCode()